component-runtime-beam/src/main/java/org/talend/sdk/component/runtime/beam/spi/BeamProducerFinder.java

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright (C) 2006-2022 Talend Inc. - www.talend.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.talend.sdk.component.runtime.beam.spi;
+
+import java.io.ObjectStreamException;
+import java.io.Serializable;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Queue;
+import java.util.concurrent.ArrayBlockingQueue;
+
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.talend.sdk.component.api.record.Record;
+import org.talend.sdk.component.api.service.source.ProducerFinder;
+import org.talend.sdk.component.runtime.base.Delegated;
+import org.talend.sdk.component.runtime.base.LifecycleImpl;
+import org.talend.sdk.component.runtime.input.Input;
+import org.talend.sdk.component.runtime.input.Mapper;
+import org.talend.sdk.component.runtime.manager.service.ProducerFinderImpl;
+import org.talend.sdk.component.runtime.manager.service.api.ComponentInstantiator;
+import org.talend.sdk.component.runtime.serialization.SerializableService;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class BeamProducerFinder extends ProducerFinderImpl {
+
+    static final Queue<Record> QUEUE = new ArrayBlockingQueue<>(20, true);
+
+    @Override
+    public Iterator<Record> find(final String familyName, final String inputName, final int version,
+            final Map<String, String> configuration) {
+        final ComponentInstantiator instantiator = getInstantiator(familyName, inputName);
+        final Mapper mapper = findMapper(instantiator, version, configuration);
+        try {
+            final Input input = mapper.create();
+            return iterator(input);
+        } catch (Exception e) {
+            log.warn("Component Kit Mapper instantiation failed, trying to wrap native beam mapper...");
+            final Object delegate = Delegated.class.cast(mapper).getDelegate();
```

Comment:
Is it Mapper or Input which contains delegated PTransform ?
(And exception occurs in iterator(input) call, so; all change could have been put here ?)
BTW, i can't find where TCK treat PTransform connectors.

Reply:
That's the mapper.
Don't get your final comment...

Reply:
ok, i see, name of S3Input class on cloud-component get me wrong, it's indeed a mapper
```java
@PartitionMapper(name = "S3Input")
@Documentation("This component reads data from S3.")
public class S3Input extends PTransform<PBegin, PCollection<Record>> {
```